protocol-tokio/src/protocol/codec/message.rs

```diff
@@ -351,63 +369,43 @@ impl de::Deserialize for GetBlocks {
     }
 }
 
-pub type Block = block::block::Block;
-
-#[cfg(test)]
-fn random_headerhash<G: ::quickcheck::Gen>(g: &mut G) -> HeaderHash {
-    let bytes: Vec<u8> = ::quickcheck::Arbitrary::arbitrary(g);
-    HeaderHash::new(&bytes)
-}
-
-#[cfg(test)]
-fn random_to<G: ::quickcheck::Gen>(g: &mut G) -> Option<HeaderHash> {
-    let value: Option<()> = ::quickcheck::Arbitrary::arbitrary(g);
-    value.map(|_| random_headerhash(g))
-}
-
-#[cfg(test)]
-fn random_from<G: ::quickcheck::Gen>(g: &mut G) -> Vec<HeaderHash> {
-    let num: usize = ::quickcheck::Arbitrary::arbitrary(g);
-    ::std::iter::repeat_with(|| random_headerhash(g))
-        .take(num)
-        .collect()
-}
-
 #[cfg(test)]
-impl ::quickcheck::Arbitrary for GetBlockHeaders {
+impl<T: ::quickcheck::Arbitrary> ::quickcheck::Arbitrary for GetBlockHeaders<T> {
```

Comment:
Move these under `mod test`. It does not matter where in the crate an impl is defined and organizationally they belong there.